ptf_tests/common/utils/test_utils.py

```diff
@@ -1021,3 +1041,160 @@ def host_check_netperf(remote=False, hostname="",username="",password=""):
+
+    vm = config_data['vm'][vm_id]
+    port = config_data['port'][vm_id]
+    
+    num = config_data['traffic']['number_pkts'][0]
+    eth_src = '\"%s\"' % port['mac'] 
+    eth_dst = '\"%s\"' % vm['mac_remote']
+    ip_src = '\"%s\"' % vm['dst_gw']
+    ip_dst= '\"%s\"' % vm['remote_ip']
+    iface = '%s' % port['interface']
+    pcap_file= config_data['traffic']['pcap_file_name'][0]
+    vm_name = vm['name']
+
+    if traffic_type=='multicast':
+        ip_dst= '\"%s\"' % config_data['traffic']['in_pkt_header']['ip_dst'][1]
+        eth_dst='\"01:00:5e:00:00:01\"'
+    elif traffic_type=='broadcast':
+        ip_dst= '\"%s\"' % config_data['traffic']['in_pkt_header']['ip_dst'][2]
+        eth_dst='\"FF:FF:FF:FF:FF\"'
+    
+    #STEP1: starting TCPDUMP on remote_vm/receiver_vm using tcpdump
+    print(f"STEP1: Starting TCPDUMP on receiver_vm pcap file: {pcap_file}")
+    tcpdump_cmd = "tcpdump -i %s -w %s" % (iface, pcap_file)
+    remote_conn.sendCmd(tcpdump_cmd)
+
+    #packet build    
+    command_list = ['python3','import scapy.all as scapy']
+
+    for cmd in command_list:
+        status = conn.sendCmd(cmd)
+        if status:
+            print(f"Command: {cmd} executed ")
+        else:
+            print("Failed to execute command {cmd}")
+
+    print(f"#######Sending Scapy Traffic from the sender_vm {vm_name} ########")
+    # Packet Building using scapy
+    conn.sendCmd('pkt=scapy.Ether(dst=%s,src=%s)/scapy.IP(dst=%s,src=%s)/\
+                  scapy.TCP()/scapy.Raw(\'0\'*50)' %(eth_dst,eth_src,ip_dst,ip_src))
+
+
+    print('Sending scapy traffic on iface=\"%s\",count=%s' % (iface,num))
+    status = conn.sendCmd('scapy.sendp(pkt,iface=\"%s\",count=%s)' % (iface,num))
+    # status = conn.sendCmd(cmd)
+    if status:
+        print(f"Command: scapy send executed from {vm_name} ")
+    else:
+        print("Failed to execute command scapy send")
+
+    result = conn.readResult()
+    if result:
+        match = re.search('Sent (\d*) packets', result)
+        if match:
+            pkt_sent = int(match.group(1))
+
+    ##Killing python3 terminal
+    status = conn.sendCmd('exit()')
+    if not status:
+        print("Failed to exit from python prompt")
+
+    if f"Sent {num} packets" in result:
+        print(f"PASS: Sent {num} to dst_ip: {ip_dst}")
+        print(f"result:{result}")
+        return True
+    else:
+        print(f"FAIL: Sent Failed--> {result} sent, expected {num} packets to be sent to dst_ip: {ip_dst}")
+        return False
+
+
+def verify_scapy_traffic_from_vm(vm_id,conn,config_data):
+    """
+    To verify packets captured on pcap file at receiver vm
+    STEPS:
+    1. Verify if pcap file exists on receiver_vm
+    2. use scapy.sniff to decode packets on the pcap file
+    3. verify the pkt_count on pcap file == expected_pkt_count (count)
+    """
+
+    result = False
+    packets = []
+    count = config_data['traffic']['number_pkts'][0]
+    pcap_file = config_data['traffic']['pcap_file_name'][0]
```

Comment:
check pcap file name.  Log shows both runs are log1.pcap.  Should have different file name for each run.  If not, delete previous one before the next run.

Reply:
logfile is been sent from config file     "traffic": 
     {
             "type": "tcp",
             "send_port": [1, 0, 0],
             "receive_port": [0, 1, 1],
             "in_pkt_header": {
       "ip_src": ["2.2.2.2"],
       "ip_dst": ["1.1.1.1", "224.0.0.1", "255.255.255.255"]
      },
            "out_pkt_header": [],
            "number_pkts": [10],
            "pcap_file_name": ["log1.pcap"]
            },